discretisedfield/tests/test_field.py

```diff
@@ -843,20 +843,27 @@ def test_mean():
         f.mean(direction=["x", "y", "z", "z"])
 
 
-# TODO Martin
-def test_field_component(valid_mesh):
-    f = df.Field(valid_mesh, nvdim=3, value=(1, 2, 3))
-    assert all(isinstance(getattr(f, i), df.Field) for i in "xyz")
-    assert all(getattr(f, i).nvdim == 1 for i in "xyz")
+@pytest.mark.parametrize("nvdim", [1, 2, 3, 4])
+def test_field_component(valid_mesh, nvdim):
+    valid_components = ["a", "b", "c", "d", "e", "f"]
+    f = df.Field(valid_mesh, nvdim=nvdim, vdims=valid_components[:nvdim])
+    assert all(isinstance(getattr(f, i), df.Field) for i in valid_components[:nvdim])
+    assert all(getattr(f, i).nvdim == 1 for i in valid_components[:nvdim])
 
-    f = df.Field(valid_mesh, nvdim=2, value=(1, 2))
-    assert all(isinstance(getattr(f, i), df.Field) for i in "xy")
-    assert all(getattr(f, i).nvdim == 1 for i in "xy")
+    # Default
+    f = df.Field(valid_mesh, nvdim=nvdim)
+    if nvdim > 3:
+        valid_components = [f"v{i}" for i in range(nvdim)]
+    elif nvdim > 1:
```

Comment:
A small design choice: to me it looks more readable if we have:
```python
if nvdim in [2, 3]:
    ...
elif nvdim > 3:
    ...
else:
    ...
```
But it is a personal choice. You can keep the tests as it is since the logic works :).

Reply:
Cheers! I've changed it.